### PR TITLE
Create documentation surrounding telemetry collection

### PR DESCRIPTION
**Pull request by IsaacPD**
**Description**
Created a generic version of hack/schemas/main.go that takes in two parameters, a go file and an output file. It reads the structs from the go file and parses any markdown in the comments into valid html and outputs it to the specified location.

This generated json is then used to populate the telemetry page.

**User facing changes**
A new page, https://skaffold.dev/docs/resources/telemetry, describing the data we collect

### CODE REVIEW

File: docs/content/en/docs/telemetry/metrics.json [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        }
+      },
+      "description": "describes how an iteration and started and if an error happened.",
+      "x-intellij-html-description": "<p>describes how an iteration and started and if an error happened.</p>"
+    },
+    "skaffoldMeter": {
+      "properties": {
+        "Arch": {
+          "description": "The architecture running skaffold as returned from runtime.GOARCH.",
+          "x-intellij-html-description": "<p>The architecture running skaffold as returned from runtime.GOARCH.</p>"
```
Comment (tejal29) [inline docs/content/en/docs/telemetry/metrics.json:31]: nit: The architecture running skaffold as returned from `runtime.GOARCH` ?

File: docs/content/en/docs/telemetry/metrics.json [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        },
+        "Build Artifacts": {
+          "description": "BuildArtifacts The number of artifacts built in the current execution as defined in skaffold.yaml.",
```
Comment (tejal29) [inline docs/content/en/docs/telemetry/metrics.json:34]:
```suggestion
          "description": "BuildArtifacts is the number of artifacts built in the current execution as defined in skaffold.yaml.",
```

File: docs/content/en/docs/telemetry/metrics.json [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          "x-intellij-html-description": "<p>BuildArtifacts The number of artifacts built in the current execution as defined in skaffold.yaml.</p>"
+        },
+        "Builders": {
+          "description": "All the builders used to build the artifacts built.",
+          "x-intellij-html-description": "<p>All the builders used to build the artifacts built.</p>"
+        },
+        "Command": {
+          "description": "The command that is used to execute skaffold `dev, build, render, run, etc.`.",
```
Comment (tejal29) [inline docs/content/en/docs/telemetry/metrics.json:42]:
```suggestion
          "description": "The command that is used to execute skaffold `dev, build, render, run, etc.` without any command line arguments",
```

File: docs/content/en/docs/telemetry/metrics.json [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        "Arch": {
+          "description": "The architecture running skaffold as returned from runtime.GOARCH.",
+          "x-intellij-html-description": "<p>The architecture running skaffold as returned from runtime.GOARCH.</p>"
+        },
+        "Build Artifacts": {
+          "description": "BuildArtifacts The number of artifacts built in the current execution as defined in skaffold.yaml.",
+          "x-intellij-html-description": "<p>BuildArtifacts The number of artifacts built in the current execution as defined in skaffold.yaml.</p>"
+        },
+        "Builders": {
+          "description": "All the builders used to build the artifacts built.",
```
Comment (tejal29) [inline docs/content/en/docs/telemetry/metrics.json:38]:
```suggestion
          "description": "Enum values for all the builders used to build the artifacts built.",
```

File: docs/content/en/docs/telemetry/metrics.json [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        "Dev Iterations": {
+          "description": "DevIterations The error results of the various dev iterations and the reasons they were triggered. The triggers can be one sync, build or deploy.",
+          "x-intellij-html-description": "<p>DevIterations The error results of the various dev iterations and the reasons they were triggered. The triggers can be one sync, build or deploy.</p>"
+        },
+        "Duration": {
+          "description": "of skaffold instance.",
+          "x-intellij-html-description": "<p>of skaffold instance.</p>"
+        },
+        "Enum Flags": {
+          "description": "EnumFlags Any flags passed into Skaffld that have a pre-defined list of valid values e.g. `'–cache-artifacts=false', '–mute-logs=[\"build\", \"deploy\"]'`.",
```
Comment (tejal29) [inline docs/content/en/docs/telemetry/metrics.json:58]:
```suggestion
          "description": "EnumFlags: enum values for  flags passed into Skaffld that have a pre-defined list of valid values e.g. `'–cache-artifacts=false', '–mute-logs=[\"build\", \"deploy\"]'`.",
```

**Review by tejal29 — CHANGES_REQUESTED** [review]
1. Move Telemetry into "Resources", 
2. Sample json data. 
3. Mention the skaffold version in which we would be collecting. v1.19.0 onwards.

We don't mention whether users are opted in by default. or opted out. 
add "It does not include argument values, or any personal information."

File: docs/content/en/docs/telemetry/metrics.json (hunk 1 above)
Comment (IsaacPD) [reply docs/content/en/docs/telemetry/metrics.json]: done.

File: docs/content/en/docs/telemetry/metrics.json (hunk 2 above)
Comment (IsaacPD) [reply docs/content/en/docs/telemetry/metrics.json]: done.

File: docs/content/en/docs/telemetry/metrics.json (hunk 3 above)
Comment (IsaacPD) [reply docs/content/en/docs/telemetry/metrics.json]: done.

File: docs/content/en/docs/telemetry/metrics.json (hunk 4 above)
Comment (IsaacPD) [reply docs/content/en/docs/telemetry/metrics.json]: done.

File: docs/content/en/docs/telemetry/metrics.json (hunk 5 above)
Comment (IsaacPD) [reply docs/content/en/docs/telemetry/metrics.json]: done,

File: docs/content/en/docs/resources/telemetry/_index.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+skaffold dev -v trace --port-forward --cache-artifacts=false --filename=./skaffold.yaml
+```
+Running the above in the [microservices example](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/microservices)
+after a couple of builds/deploys results in the following metrics being collected
+```json
+[{
+    "ExitCode": 0,
+    "BuildArtifacts": 3,
+    "Command": "dev",
+    "Version": "v1.18.0-test",
```
Comment (tejal29) [inline docs/content/en/docs/resources/telemetry/_index.md:27]: Change version to v1.19.0

File: docs/content/en/docs/resources/telemetry/_index.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+title: "Telemetry"
+linkTitle: "Telemetry"
+weight: 400
+aliases: [/docs/metrics]
+---
+
+<script type="module" src="main.js"></script>
+
+To help improve the quality of this product, we collect anonymized usage data
+from Skaffold by default on version 1.19.0 and onward.
```
Comment (tejal29) [inline docs/content/en/docs/resources/telemetry/_index.md:11]:
```suggestion
To help improve the quality of this product, we collect anonymized usage data from Skaffold.

You will be *opted-in* by default and you can opt-out anytime via a `skaffold config` command

Usage data does not include argument values, or any personal information.

 In order to disable sending usage data, run the following command after you have installed skaffold:

```
skaffold config set --global collect-metrics false
```

File: docs/content/en/docs/resources/telemetry/_index.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+}]
+```
+
+This data is handled in accordance with our privacy policy [https://policies.google.com/privacy](https://policies.google.com/privacy).
+
+
+You may choose to opt out of this collection by running the following command:
+```bash
+skaffold config set --global collect-metrics false
+```
```
Comment (tejal29) [inline docs/content/en/docs/resources/telemetry/_index.md:68]:
```suggestion
```

**Review by tejal29 — COMMENTED** [review]
final nits

**Review by tejal29 — APPROVED** [review]

**Pull request merged**